scrapy/pipelines/files.py

```diff
@@ -248,6 +250,50 @@ def persist_file(self, path, buf, info, meta=None, headers=None):
         )
 
 
+class FTPFilesStore(object):
+
+    FTP_USERNAME = None
+    FTP_PASSWORD = None
+    
+    def __init__(self, uri):
+        assert uri.startswith('ftp://')
+        u = urlparse(uri)
+        self.ftp = FTP()
+        self.ftp.connect(u.hostname, u.port or '21')
+        username = u.username or FTP_USERNAME
+        password = u.password or FTP_PASSWORD
+        self.ftp.login(username, password)
+        self.basedir = u.path
+        
+    def persist_file(self, path, buf, info, meta=None, headers=None):
+        buf.seek(0)
+        # If the path is like 'x/y/z.ext' the 'x/y' is rel_path and 
+        # 'z.ext' is file name
+        # If path is only the file name 'z.ext', then rel_path is
+        # the empty string and filename is 'z.ext'
+        x = path.rsplit('/',1)
+        rel_path, filename = ('/' + x[0], x[1]) if len(x) > 1 else ('', x[0])
+        abs_path = self.basedir + rel_path
```

Comment:
What if basedir ends with `/`?

Reply:
'Never trust the user' is what I should have done , hahaha. Will fix that